projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -360,12 +362,80 @@ export class IgxTreeGridComponent extends IgxGridBaseDirective implements GridTy
             this.loadChildrenOnRowExpansion(args);
         });
 
+        this.onRowAdded.subscribe(args => {
+            if (this.rowSelection === 'multipleCascade') {
+                let rec = (this.gridAPI as IgxTreeGridAPIService).get_rec_by_id(this.primaryKey ? args.data[this.primaryKey] : args.data);
+                if (rec && rec.parent) {
+                    // if batch editing is enabled
+                    (this.gridAPI as IgxTreeGridAPIService).handleCascadeSelectionByFilteringAndCRUD(
+                        new Set([rec.parent]), true, undefined, rec.parent.rowID);
+                } else {
+                    // if batch editin is disabled
+                    requestAnimationFrame(() => {
```

Comment:
Is this the same as handleCascadeSelection? Kind of looks like it. Also the batch comment might need elaborating.

Reply:
Also, is the animation frame because the event emits too soon (unlike filtering?) in that case leave a comment why it's there as this will likely need to be remove in the future. Also doesn't the event itself in the tree grid provide record parent information?!